app/controllers/api/v2/anime_controller.rb

```diff
@@ -2,6 +2,21 @@ module Api::V2
   class AnimeController < ApiController
     caches_action :show, expires_in: 1.hour
 
+    def index
+      anime = Anime.preload(:genres)
+
+      if params.key? :filter
+        search = AnimeSearch.new(params[:filter])
+        if search.valid?
+          anime = search.apply_filters(anime)
+        else
+          return render json: search.errors.full_messages
+        end
+      end
+
+      render json: anime.page(params[:page]).per(20)
```

Comment:
Just found a :beetle: on this that needs fixing before anyone starts using the endpoint: this needs to be `AnimeSerializer.new(anime).as_json` (I don't believe `Api::V2` is required) like below, and we will want to include `meta` for pagination